darts/models/torch_forecasting_model.py

```diff
@@ -401,70 +405,202 @@ def predict(self,
         covariates = [covariates] if isinstance(covariates, TimeSeries) else covariates
 
         dataset = SimpleInferenceDataset(series, covariates)
-        predictions = self.predict_from_dataset(n, dataset)
+        predictions = self.predict_from_dataset(n, dataset, verbose=verbose)
         return predictions[0] if called_with_single_series else predictions
 
     def predict_from_dataset(self,
                              n: int,
-                             input_series_dataset: TimeSeriesInferenceDataset
-                             ) -> Sequence[TimeSeries]:
+                             input_series_dataset: Union[TimeSeriesInferenceDataset, torch.Tensor, np.ndarray],
```

Comment:
Unfortunately I don't think it's a good idea to accept tensors of Numpy arrays here. By doing this, we would loose a lot of the nice "all included" services that Darts provides. Users would have to worry about building tensors or arrays, which goes against the principles of Darts (the only thing users have to know are `TimeSeries`). Users that don't want to shouldn't even have to know that we are using Pytorch under the hood.

The good news is that this can be done very easily. You already built the most importance piece of the puzzle, which is the batched `_produce_prediction()` method. This is very good. Now in order to have fast inference using `TimeSeriesInferenceDataset` I think you should (roughly) follow the following steps:

* Change the `SimpleInferenceDataset.__getitem__()` so that it returns `np.ndarray` instead of `TimeSeries`. Creating all these `TimeSeries` is likely to be the time consuming thing here. Probably doing something like
```
    def __getitem__(self, idx: int) -> Tuple[np.ndarray, Optional[np.ndarray]]:
        return (self.series[idx].values(copy=False), None) if self.covariates is None else (self.series[idx].values(copy=False), self.covariates[idx].values(copy=False))
```
is enough. Accessing the values without copying will be fast because it uses Numpy views.

* Instantiate a `TimeSeriesTorchDataset` (like we are doing for training). There's nothing more to do here, it should work out of the box.

* Create your torch `DataLoader` from this `TimeSeriesTorchDataset` (that too should work out of the box).

That should remove quite a bit of code from what you currently have, but most importantly will allow users to not have to worry about tensors and array, but just about `TimeSeries`.